src/tzdata/TZData.jl

```diff
@@ -2,18 +2,18 @@ module TZData
 
 using LazyArtifacts
 using Printf
-using ...TimeZones: DEPS_DIR
+using ...TimeZones: scratch_dir
 
 # Note: The tz database is made up of two parts: code and data. TimeZones.jl only requires
 # the "tzdata" archive or more specifically the "tz source" files within the archive
 # (africa, australasia, ...)
 
-const TZ_SOURCE_DIR = joinpath(DEPS_DIR, "tzsource")
-const COMPILED_DIR = joinpath(DEPS_DIR, "compiled", string(VERSION))
+tz_source_dir() = scratch_dir("tzsource")
+serialized_cache_dir() = scratch_dir("serialized", string(VERSION))
```

Comment:
❗ We'll need to use different directories for each tzdata version. Otherwise having multiple versions of TimeZones installed at the same time with different tzdata versions will interfere with each other